dev/archery/archery/crossbow/cli.py

```diff
@@ -339,7 +340,24 @@ def asset_callback(task_name, task, asset):
             path = target_dir / task_name / asset.name
             path.parent.mkdir(exist_ok=True)
             if not dry_run:
-                asset.download(path)
+                import github3
+                max_n_retries = 5
+                n_retries = 0
+                while True:
+                    try:
+                        asset.download(path)
+                    except github3.exceptions.GitHubException as error:
+                        n_retries += 1
+                        if n_retries == max_n_retries:
```

Comment:
Using a for loop you can remove this check.

Reply:
If we use a for loop, no exception is raised when all retries are failed.

Reply:
Right, didn't think of that :)